api/mapping/models.py

```diff
@@ -354,3 +366,43 @@ class NLPModel(models.Model):
 
     def __str__(self):
         return str(self.id)
+
+
+class Dataset(BaseModel):
+    """
+    Model for datasets which contain scan reports.
+    """
+
+    id = models.UUIDField(primary_key=True, editable=False, default=uuid.uuid4)
```

Comment:
why do we need an `id` field that is `uuid4` ? What's wrong the the `django` default (from `BaseModel`) `id` field?

Reply:
The default auto-increment integer is less safe than UUIDs. If/when you get hacked, having id as 1, 2, 3 makes discoverable by iteration. Using random UUIDs stops people requesting `sensitive_item` 1-infinity

Reply:
makes sense. Was/is there a plan to migrate existing models or just these, out of interest? Just had a think and dont think that'd be a problem at all for the API / rules generation for the mapping json file - often we used the existing 'id' just as a string key

Reply:
There's not a plan so far. Something to think about though.